cron/loginstats.go

```diff
@@ -0,0 +1,70 @@
+package cron
+
+import "database/sql"
+
+type Login struct {
+	Count int
+}
+
+// /logins/distinct
+func GetDistinctLoginCount(db *sql.DB, startDate string, endDate string) (Login, error){
+	var count int
+
+	query := `SELECT count(distinct user_id) FROM logins WHERE login_time >= ($1 :: DATE )
+		AND login_time <= ($2 :: DATE);`
+
+	rows, err := db.Query(query, startDate, endDate)
+
+	if err != nil {
+		return Login{-1}, err
+	}
+
+	defer rows.Close()
+	var logins Login
+	logins = Login{0}
+	for rows.Next() {
+		err := rows.Scan(&count)
+		if err != nil {
+			return Login{-1}, err
+		}
+		logins.Count += count
+	}
+
+	err = rows.Err()
+	if err != nil {
+		return Login{-1}, err
+	}
+
+	return logins, nil
+}
+
+// /login
+func GetLoginCount(db *sql.DB, startDate string, endDate string) (Login, error){
+	var count int
+
+	query := `select count(user_id) from logins where login_time >= ($1 :: DATE) 
+		AND login_time <= ($2 :: DATE);`
+
+	rows, err := db.Query(query, startDate, endDate)
```

Comment:
This might be best taken with a grain of salt because I don't like this function much, but `database/sql` does have a convenience function for when only one row is returned, `QueryRow`. The interface is subtly different than with `Query` and the `Rows` type, but it might make the code look nicer. See https://golang.org/pkg/database/sql/#DB.QueryRow